django_fsm_log/migrations/0004_add_source_state.py

```diff
@@ -0,0 +1,24 @@
+# Generated by Django 2.1.1 on 2018-09-04 10:35
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('django_fsm_log', '0003_statelog_description'),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name='statelog',
+            name='source_state',
+            field=models.CharField(db_index=True, default='', max_length=255),
```

Comment:
default of None here maybe, i.e. `null=True` (and `blank=True`) in models?

Reply:
@jonasvp 
I'd be happy to push a fixup myself if you agree that it makes sense.

My point here is that all existing entries should rather be NULL than `''`.

Reply:
Anway, please rebase it on master then if you push a fix.

Reply:
Well, the Django docs advise against using `null=True` on `CharFields` (https://docs.djangoproject.com/en/2.1/ref/models/fields/#null):

> Avoid using null on string-based fields such as CharField and TextField. If a string-based field has null=True, that means it has two possible values for “no data”: NULL, and the empty string.

What problem would setting `null=True` solve?

As for `blank=True`: it probably wouldn't hurt in case the source state is empty but it only applies to the admin and a `ModelForm`. Since this model only gets created programmatically, I'm not sure it would serve a purpose.

Reply:
> it has two possible values for “no data”: NULL, and the empty string.

I think that it is good here, since NULL means: migrated/unknown/unset, and the empty string means `state=''` (not really used in practice probably, but a valid use case after all).

I think the Django docs are more meant for novices, and it is good here to distinguish between unset and "empty state value".

Re `blank=True`: you might expose the model in the admin or somewhere else where a modelform is used, and then you could not use an empty value there.
In general I think `blank=True` should be used with `null=True`.

If `source_state` is empty/missing later it should be also NULL/None then probably, instead of an empty string.  But the empty string for source state should be logged as such.

This mainly depends on if `state=''` is valid after all, but even if not, using NULL for migrated values is good IMHO.